diff --git a/src/model/outputChannel.ts b/src/model/outputChannel.ts
index 9513e494..c18625cc 100644
--- a/src/model/outputChannel.ts
+++ b/src/model/outputChannel.ts
@@ -1,5 +1,4 @@
 import { Buffer, Neovim } from '@chemzqm/neovim'
-import { constants } from 'buffer'
 import { Disposable } from 'vscode-languageserver-protocol'
 import { OutputChannel } from '../types'
 import { disposeAll } from '../util'
@@ -41,7 +40,7 @@ export default class BufferChannel implements OutputChannel {
   }
 
   public append(value: string): void {
-    if (this._content.length + value.length >= constants.MAX_STRING_LENGTH) {
+    if (this._content.length + value.length >= 1073741799) {
       this.clear(10)
     }
     this._content += value
@@ -51,7 +50,7 @@ export default class BufferChannel implements OutputChannel {
   }
 
   public appendLine(value: string): void {
-    if (this._content.length + value.length >= constants.MAX_STRING_LENGTH) {
+    if (this._content.length + value.length >= 1073741799) {
       this.clear(10)
     }
     this._content += value + '\n'
