diff --git a/esbuild.js b/esbuild.js
index 4bffe41b..fc823fd9 100755
--- a/esbuild.js
+++ b/esbuild.js
@@ -77,7 +77,7 @@ async function start(watch) {
     platform: 'node',
     target: 'node10.12',
     outfile: 'build/index.js',
-    plugins: [entryPlugin, envPlugin]
+    plugins: [entryPlugin]
   })
 }
 
@@ -96,5 +96,6 @@ if (process.argv.length > 2 && process.argv[2] === '--watch') {
 }
 
 start(watch).catch(e => {
+  process.exitCode = 1
   console.error(e)
 })
