diff --git a/esbuild.js b/esbuild.js
index b1a89427..7468e464 100755
--- a/esbuild.js
+++ b/esbuild.js
@@ -41,7 +41,7 @@ async function start(watch) {
     platform: 'node',
     target: 'node10.12',
     outfile: 'build/index.js',
-    plugins: [envPlugin]
+    plugins: []
   })
 }
 
@@ -60,5 +60,6 @@ if (process.argv.length > 2 && process.argv[2] === '--watch') {
 }
 
 start(watch).catch(e => {
+  process.exitCode = 1
   console.error(e)
 })
