diff --git a/docs/users_guide/flags.py b/docs/users_guide/flags.py
index cc30b8c..a6efcbe 100644
--- a/docs/users_guide/flags.py
+++ b/docs/users_guide/flags.py
@@ -600,11 +600,11 @@ def setup(app):
 
     # Add ghc-flag directive, and override the class with our own
     app.add_object_type('ghc-flag', 'ghc-flag')
-    app.add_directive_to_domain('std', 'ghc-flag', Flag)
+    app.add_directive_to_domain('std', 'ghc-flag', Flag, override=True)
 
     # Add extension directive, and override the class with our own
     app.add_object_type('extension', 'extension')
-    app.add_directive_to_domain('std', 'extension', LanguageExtension)
+    app.add_directive_to_domain('std', 'extension', LanguageExtension, override=True)
     # NB: language-extension would be misinterpreted by sphinx, and produce
     # lang="extensions" XML attributes
 
