diff --git a/runtime/autoload/health/nvim.vim b/runtime/autoload/health/nvim.vim
index c25f5ee64..6be2b6183 100644
--- a/runtime/autoload/health/nvim.vim
+++ b/runtime/autoload/health/nvim.vim
@@ -4,13 +4,13 @@ function! s:check_config() abort
   let ok = v:true
   call health#report_start('Configuration')
 
-  let vimrc = empty($MYVIMRC) ? stdpath('config').'/init.vim' : $MYVIMRC
-  if !filereadable(vimrc)
-    let ok = v:false
-    let has_vim = filereadable(expand('~/.vimrc'))
-    call health#report_warn((-1 == getfsize(vimrc) ? 'Missing' : 'Unreadable').' user config file: '.vimrc,
-          \[ has_vim ? ':help nvim-from-vim' : ':help init.vim' ])
-  endif
+  " let vimrc = empty($MYVIMRC) ? stdpath('config').'/init.vim' : $MYVIMRC
+  " if !filereadable(vimrc)
+  "  let ok = v:false
+  "  let has_vim = filereadable(expand('~/.vimrc'))
+  "  call health#report_warn((-1 == getfsize(vimrc) ? 'Missing' : 'Unreadable').' user config file: '.vimrc,
+  "        \[ has_vim ? ':help nvim-from-vim' : ':help init.vim' ])
+  " endif
 
   " If $VIM is empty we don't care. Else make sure it is valid.
   if !empty($VIM) && !filereadable($VIM.'/runtime/doc/nvim.txt')
@@ -170,9 +170,9 @@ function! s:check_tmux() abort
     call health#report_error(
           \ '$TERM differs from the tmux `default-terminal` setting. Colors might look wrong.',
           \ ['$TERM may have been set by some rc (.bashrc, .zshrc, ...).'])
-  elseif $TERM !~# '\v(tmux-256color|screen-256color)'
+  elseif $TERM !~# '\v(tmux-256color|screen-256color|xterm-kitty)'
     call health#report_error(
-          \ '$TERM should be "screen-256color" or "tmux-256color" in tmux. Colors might look wrong.',
+          \ '$TERM should be "xterm-kitty", "screen-256color" or "tmux-256color"  in tmux. Colors might look wrong.',
           \ ["Set default-terminal in ~/.tmux.conf:\nset-option -g default-terminal \"screen-256color\"",
           \  s:suggest_faq])
   endif
